src/Sarif/Core/ArtifactLocation.cs

```diff
@@ -20,6 +20,11 @@ public partial class ArtifactLocation
         /// <returns></returns>
         public bool TryReconstructAbsoluteUri(IDictionary<string, ArtifactLocation> originalUriBaseIds, out Uri resolvedUri)
         {
+            resolvedUri = null;
+
+            // If this artifactLocation represents a top level originalUriBaseId, then the value of the URI may be absent.
+            if (this.Uri == null) { return false; }
+
             resolvedUri = this.Uri.IsAbsoluteUri ? this.Uri : null;
```

Comment:
>resolvedUri [](start = 12, length = 11)

If `this.Uri.IsAbsoluteUri`, then we are done, regardless of what `originalUriBaseIds` does or does not say. So replace this line with:
```
if (this.Uri.IsAbsoluteUri)
{
    resolvedUri = this.Uri;
    return true;
}
```